tracing/src/main/java/com/palantir/tracing/TraceLocalObserver.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tracing;
+
+import javax.annotation.Nullable;
+
+public interface TraceLocalObserver<T> {
+    void onTraceComplete(String traceId, @Nullable String requestId, T value);
```

Comment:
We could make this a default method with a no-op implementation, retaining the potential for other observer methods in the future in a consistent way.

This method needs javadoc describing what it's measuring.

Reply:
made default

javadoc incoming...